### PR TITLE
Add eol parameter for NormalizeSyntax API

### LINKED ISSUE #3318: The NormalizeSyntax API should let you specify the type of EOL to use.
The NormalizeSyntax API already lets you specify the whitespace to use for each indentation level.  It should also let you specify the end-of-line whitespace, as opposed to always using CRLF.

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #3318 

This changes adds an optional "eol" parameter to the NormalizeWhitespace API that lets you specify the text to use for end of line.  This is a public API change that should be binary & recompile compatible with the existing API (except for a few protected 'core' API's that none one outside of Roslyn should be implementing)

@AnthonyDGreen @gafter @Pilchie please review.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/PublicAPI.txt [hunk 1]
```diff
@@ -3389,8 +3389,10 @@ static Microsoft.CodeAnalysis.CSharp.SymbolDisplay.ToDisplayParts(Microsoft.Code
 static Microsoft.CodeAnalysis.CSharp.SymbolDisplay.ToDisplayString(Microsoft.CodeAnalysis.ISymbol symbol, Microsoft.CodeAnalysis.SymbolDisplayFormat format = null) -> string
 static Microsoft.CodeAnalysis.CSharp.SymbolDisplay.ToMinimalDisplayParts(Microsoft.CodeAnalysis.ISymbol symbol, Microsoft.CodeAnalysis.SemanticModel semanticModel, int position, Microsoft.CodeAnalysis.SymbolDisplayFormat format = null) -> System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.SymbolDisplayPart>
 static Microsoft.CodeAnalysis.CSharp.SymbolDisplay.ToMinimalDisplayString(Microsoft.CodeAnalysis.ISymbol symbol, Microsoft.CodeAnalysis.SemanticModel semanticModel, int position, Microsoft.CodeAnalysis.SymbolDisplayFormat format = null) -> string
-static Microsoft.CodeAnalysis.CSharp.SyntaxExtensions.NormalizeWhitespace(this Microsoft.CodeAnalysis.SyntaxToken token, string indentation = "    ", bool elasticTrivia = false) -> Microsoft.CodeAnalysis.SyntaxToken
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/PublicAPI.txt]: This appears to be a breaking change, so it isn't OK.

Comment (gafter) [inline src/Compilers/CSharp/Portable/PublicAPI.txt]: This is a breaking API change. We should not do that.

Comment (mattwar) [inline src/Compilers/CSharp/Portable/PublicAPI.txt]: It is still binary and source compatible.  This signature is removed and replace with one with exact same parameters, but not optional.

File: src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions.cs [hunk 2]
```diff
@@ -277,6 +277,7 @@ public static TRoot ReplaceTrivia<TRoot>(this TRoot root, SyntaxTrivia trivia, S
         }
 
         internal const string DefaultIndentation = "    ";
+        internal const string DefaultEOL = "\r\n";
```
Comment (gafter) [inline src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions.cs]: It is a bit annoying that the API's default line-ending-sequence on Linux is the Windows sequence.

**Comment by gafter** [comment]
LGTM

File: src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions.cs (hunk 2 above)
Comment (mattwar) [inline src/Compilers/Core/Portable/Syntax/SyntaxNodeExtensions.cs]: That's the way the current implementation is.   If we want to change the behavior we can think about that separately.

**Comment by mattwar** [comment]
@pilchie any comments?

**Comment by Pilchie** [comment]
The Core change is breaking, if there is a way for someone to make a SyntaxNode (I forget whether that's possible or not).

Also in case the three named args are specified, overload resolution picks the old method, not an ambiguity, right?

Finally, are you intending this for 1.0?

**Comment by mattwar** [comment]
@pilchie  It is not for 1.0. No one currently can make a sub type of SyntaxNode, so the change to a protected member shouldn't matter. There is no ambiguity for the language to pick one of the methods.

**Comment by Pilchie** [comment]
Okay - you've convinced me.  :+1:

**Pull request merged**